File: assignment/269p3-4.R
pima <- read.csv('./data/pima2.csv', header = T)

classify <- function(pregnant) {
    if (pregnant <= 5) {
        return('0~5')
    }else if (6 <= pregnant && pregnant <= 10) {
        return('6~10')
    }else {
        return('10+')
    }
}

pima$pregnant <- unlist(lapply(pima$pregnant, classify))
tbl <- table(pima$pregnant, pima$diabetes)
tbl
barplot(t(tbl), col = 1:2, legend = colnames(tbl))

File: assignment/269p3-4.py
import pandas as pd
import matplotlib.pyplot as plt

pima = pd.read_csv('../data/pima2.csv')


def classify(pregnant):
    if pregnant <= 5:
        return '0~5'
    elif 6 <= pregnant <= 10:
        return '6~10'
    else:
        return '10+'


pima['pregnant'] = pima['pregnant'].apply(classify)
table = pd.crosstab(index=pima['pregnant'], columns=pima['diabetes'])
print(table)
table.plot.bar(stacked=True)
plt.show()
